### PR TITLE
feat: add new casts search route to SDK

### PR DESCRIPTION
**Pull request by valerierose**
Adds `cast/search` to SDK.

Also pulls in some recent changes to the OAS.

### CODE REVIEW

File: src/neynar-api/neynar-api-client.ts [hunk 1]
```diff
@@ -1538,6 +1538,42 @@ export class NeynarAPIClient {
...
+   * @returns {Promise<CastsResponse>} A promise that resolves to a `CastsResponse` object,
+   *   containing the results of the casts search.
+   *
+   * @example
+   * // Example: Search for casts with a specific query
+   * client.searchCasts('star wars').then(response => {
+   *   console.log('Casts Search Results:', response); // Outputs the results of the casts search
+   * });
+   *
+   * For more information, refer to the [Neynar documentation](https://docs.neynar.com/reference/casts-search).
```
Comment (MrKevinOConnell) [inline src/neynar-api/neynar-api-client.ts:1562]: https://docs.neynar.com/reference/cast-search

File: src/neynar-api/v2/client.ts [hunk 2]
```diff
@@ -1115,6 +1115,51 @@ export class NeynarV2APIClient {
...
+   * @returns {Promise<CastsResponse>} A promise that resolves to a `CastsResponse` object,
+   *   containing the results of the casts search.
+   *
+   * @example
+   * // Example: Search for casts with a specific query
+   * client.searchCasts('star wars').then(response => {
+   *   console.log('Casts Search Results:', response); // Outputs the results of the casts search
+   * });
+   *
+   * For more information, refer to the [Neynar documentation](https://docs.neynar.com/reference/casts-search).
```
Comment (MrKevinOConnell) [inline src/neynar-api/v2/client.ts:1139]: https://docs.neynar.com/reference/cast-search

**Review by MrKevinOConnell — CHANGES_REQUESTED** [review]
one small thing

File: src/neynar-api/v2/openapi-farcaster/apis/frame-api.ts [hunk 3]
```diff
@@ -550,6 +596,18 @@ export const FrameApiFp = function(configuration?: Configuration) {
...
         },
+        /**
+         * Fetches the frame meta tags from the URL
+         * @summary Fetches the frame meta tags from the URL
+         * @param {string} apiKey API key required for authentication.
+         * @param {string} url The frame URL to crawl
+         * @param {*} [options] Override http request option.
+         * @throws {RequiredError}
+         */
+        async frameFromUrl(apiKey: string, url: string, options?: AxiosRequestConfig): Promise<(axios?: AxiosInstance, basePath?: string) => AxiosPromise<FrameFromUrl200Response>> {
```
Comment (Shreyaschorge) [inline src/neynar-api/v2/openapi-farcaster/apis/frame-api.ts:607]: Need wrapper code for `frameFromUrl`

File: src/neynar-api/v2/client.ts [hunk 4]
```diff
@@ -1115,6 +1115,51 @@ export class NeynarV2APIClient {
...
+   *
+   * @example
+   * // Example: Search for casts with a specific query
+   * client.searchCasts('star wars').then(response => {
+   *   console.log('Casts Search Results:', response); // Outputs the results of the casts search
+   * });
+   *
+   * For more information, refer to the [Neynar documentation](https://docs.neynar.com/reference/cast-search).
+   */
+  public async searchCasts(
```
Comment (Shreyaschorge) [inline src/neynar-api/v2/client.ts:1141]: OAS change:
500 should also be an expected response

File: src/neynar-api/v2/openapi-farcaster/apis/user-api.ts [hunk 5]
```diff
@@ -592,6 +594,43 @@ export const UserApiAxiosParamCreator = function (configuration?: Configuration)
...
+            };
+        },
+        /**
+         * Fetches power users and respond in a backwards compatible format to Warpcast\'s deprecated power badge endpoint.
+         * @summary Fetch power user FIDs
+         * @param {string} apiKey API key required for authentication.
+         * @param {*} [options] Override http request option.
+         * @throws {RequiredError}
+         */
+        userPowerLite: async (apiKey: string, options: AxiosRequestConfig = {}): Promise<RequestArgs> => {
```
Comment (Shreyaschorge) [inline src/neynar-api/v2/openapi-farcaster/apis/user-api.ts:613]: Need wrapper code for `userPowerLite`

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]
Need wrapper code for 2 methods.
Rest looks good.

File: src/neynar-api/v2/openapi-farcaster/apis/user-api.ts (hunk 5 above)
Comment (eurbs) [reply src/neynar-api/v2/openapi-farcaster/apis/user-api.ts:613]: I'll add this wrapper code in a separate PR on Monday 👍

**Comment by zacharytyhacz** [comment]
Thank you sir

**Comment by zacharytyhacz** [comment]
This does not appear to be working for me, I am trying 

```typescript
await neynar.searchCasts('tabletop', { channelId: 'intori' })

// and 
await neynar.searchCasts('olympics', { channelId: 'intori' })

```

And the result is always empty, even though casts with these terms do exist:
image

image

**Comment by zacharytyhacz** [comment]
please fix @valerierose

**Comment by zacharytyhacz** [comment]
Bump

**Comment by zacharytyhacz** [comment]
Bump

**Pull request merged**